waiter/src/waiter/auth/authentication.clj

```diff
@@ -26,12 +27,26 @@
   (wrap-auth-handler [this request-handler]
     "Attaches middleware that enables the application to perform authentication.
      The middleware should
-     - either issue a 401 challenge asking the client to authenticate itself,
-     - or upon successful authentication populate the request with :authorization/user and :authorization/principal"))
+     - issue a 401 challenge, or redirect, to get the client to authenticate itself,
+     - or upon successful authentication populate the request with :authorization/user and :authorization/principal")
+  (get-authentication-providers [this]
+    "Get a list of supported authenticaiton provider names.")
+  (process-callback [this request]
+    "Process any requests that might come in after initiating authentication. e.g. receive a request
+     with an authentication assertion after redirecting a user to authenticate with an identity provider."))
+
+(extend-protocol Authenticator
+  Object
+  (get-authentication-providers [_] [])
```

Comment:
Should this list include `"standard"`? Also, should this function return a set?

Reply:
`composite` returns `standard`, for consistency so should all other authenticators. We can remove `standard` from `core` and then consider avoid explicitly mentioning `standard` in parameters doc.

Reply:
I'd rather just leave both as special for now, including in the documentation.  And make a separate PR if we think the documentation/usage should be simplified

Reply:
Since we are introducing `get-authentication-providers` in this PR, it is nice to resolve it now. But we can refer to a third reviewer for another opinion.